table/block_based_table_reader.cc

```diff
@@ -1300,63 +1308,69 @@ Status BlockBasedTable::PutDataBlockToCache(
     const Slice& block_cache_key, const Slice& compressed_block_cache_key,
     Cache* block_cache, Cache* block_cache_compressed,
     const ReadOptions& /*read_options*/, const ImmutableCFOptions& ioptions,
-    CachableEntry<Block>* block, Block* raw_block, uint32_t format_version,
-    const Slice& compression_dict, size_t read_amp_bytes_per_bit, bool is_index,
-    Cache::Priority priority, GetContext* get_context,
-    MemoryAllocator* allocator) {
-  assert(raw_block->compression_type() == kNoCompression ||
+    CachableEntry<Block>* block, BlockContents* raw_block_contents,
+    CompressionType raw_block_comp_type, uint32_t format_version,
+    const Slice& compression_dict, SequenceNumber seq_no,
+    size_t read_amp_bytes_per_bit, bool is_index, Cache::Priority priority,
+    GetContext* get_context, MemoryAllocator* allocator) {
+  assert(raw_block_comp_type == kNoCompression ||
          block_cache_compressed != nullptr);
 
   Status s;
   // Retrieve the uncompressed contents into a new buffer
-  BlockContents contents;
+  BlockContents uncompressed_block_contents;
   Statistics* statistics = ioptions.statistics;
-  if (raw_block->compression_type() != kNoCompression) {
-    UncompressionContext uncompression_ctx(raw_block->compression_type(),
+  if (raw_block_comp_type != kNoCompression) {
+    UncompressionContext uncompression_ctx(raw_block_comp_type,
                                            compression_dict);
-    s = UncompressBlockContents(uncompression_ctx, raw_block->data(),
-                                raw_block->size(), &contents, format_version,
-                                ioptions, allocator);
+    s = UncompressBlockContents(
+        uncompression_ctx, raw_block_contents->data.data(),
+        raw_block_contents->data.size(), &uncompressed_block_contents,
+        format_version, ioptions, allocator);
   }
   if (!s.ok()) {
-    delete raw_block;
     return s;
   }
 
-  if (raw_block->compression_type() != kNoCompression) {
-    block->value = new Block(std::move(contents), raw_block->global_seqno(),
+  if (raw_block_comp_type != kNoCompression) {
+    block->value = new Block(std::move(uncompressed_block_contents), seq_no,
                              read_amp_bytes_per_bit,
                              statistics);  // uncompressed block
   } else {
-    block->value = raw_block;
-    raw_block = nullptr;
```

Comment:
We used to set raw_block to nullptr? Do we need to do the same with raw_block_contents?

Reply:
can we add a few lines to the PutDataBlockToCache declaration to clarify that the passed raw_block_contents object is undefined after the function call?